docs/s3.md

```diff
@@ -0,0 +1,72 @@
+# Performance Test Result storage in S3 bucket for SLV
+
+**NOTE: This document is only relevant to puppet employees**
+
+The SLV team has an s3 bucket for storing results from performance testing.
+
+`
+s3://slv-performance-results
+`
+
+## Convention
+
+S3 is [object storage](https://docs.aws.amazon.com/AmazonS3/latest/user-guide/using-folders.html),
+so it doesn't have true directories.  But it does like to make it look like it does.  The convention 
+is to use names like `foo/bar`  and `foo/car` to group files together.  They are really two separate
+file that exist next to each other, but with the cli you can `ls foo/` and you will get two
+separate files.
+
+For our use, we will put results relating to a specific release in `<release_name>/<ticket>` Using
+the internal name like Kearney.
+
+If it isn't specific to a release, put it in `tickets/<ticket>`
+ 
+## How to access/use it
+
+The bucket is owned by the Developer user for our ESO account which we should all have access
+to.  Use that from the aws console to manipulate anything about how the bucket is setup. 
+
+### Install the aws cli
+
+Follow the 
+[Installing AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-install.html) 
+instructions for the machine you want to access s3 from.
+
+### Credentials
+
+IT set up a service account for us to use when accessing the bucket via the cli.  They
+added the credentials to the 1password SLV team vault as the S3 service account login.
+ 
+
+Create a `.aws/credentials` file in your homedir.
+Something like this:
+```
+[default]
+aws_access_key_id = <personal id>
+aws_secret_access_key = <personal key>
+          
+[slv_s3_service_account]
+aws_access_key_id = <slv s3 id>
+aws_secret_access_key = <slv s3 key>
+```
+
+### CLI example
+
+```
+aws s3 ls s3://slv-performance-results/ --profile slv_s3_service_account
+```
+
```

Comment:
It might be helpful to include an example of uploading a results dir using `cp` and then `ls` that dir. I can write that up when I upload results for SLV-426.

Reply:
I was able to upload successfully if you want to include this description:

For example, to upload the result directory `/Users/bill.claytor/RubymineProjects/gplt/slv-425/PERF_1557340076` to the `releases/Kearney/SLV-425` directory use the following command:
```
aws s3 cp --recursive /Users/bill.claytor/RubymineProjects/gplt/slv-425/PERF_1557340076 s3://slv-performance-results/releases/Kearney/SLV-425/PERF_1557340076 --profile slv_s3_service_account 
```

To list the releases, use the following command (note the `/` after `releases`:
```
bill.claytor:~> aws s3 ls s3://slv-performance-results/releases/ --profile slv_s3_service_account
                           PRE Kearney/
```
To list the contents of the result directory, use the following command:
```
bill.claytor:~> aws s3 ls s3://slv-performance-results/releases/Kearney/SLV-425/PERF_1557340076/ --profile slv_s3_service_account
                           PRE 2019-05-08_11_06_40/
                           PRE bzi4xky3n1bfknm.delivery.puppetlabs.net/
                           PRE so9p0vxuljb0g1g.delivery.puppetlabs.net/
2019-05-08 12:32:17     363228 Screen Shot 2019-05-08 at 11.47.12 AM.png
```

Reply:
updated